src/TestFramework/MSTest.Core/Attributes/DataRowAttribute.cs

```diff
@@ -69,7 +69,7 @@ public string GetDisplayName(MethodInfo methodInfo, object[] data)
             {
                 if (data != null)
                 {
-                    return string.Format(CultureInfo.CurrentCulture, FrameworkMessages.DataDrivenResultDisplayName, methodInfo.Name, string.Join(",", data));
+                    return string.Format(CultureInfo.CurrentCulture, FrameworkMessages.DataDrivenResultDisplayName, methodInfo.Name, string.Join(",", (IEnumerable<object>)data));
```

Comment:
would `DataSource` need something similar too?

Reply:
No. DisplayName in case of DataSourceAttribute doesn't consist of any data as such. It looks like "TestMethod (Data Row 0)", which is irrespective of the data.